opencog/atomspace/AtomSpace.h

```diff
@@ -358,6 +374,57 @@ class AtomSpace
     }
 
     /**
+     * Gets all the atoms of any type and appends them to the HandleSeq.
+     *
+     * @param appendToHandles the HandleSeq to which to append the handles.
+     *
+     * Example of call to this method:
+     * @code
+     *         std::list<Handle> atoms;
+     *         atomSpace.get_all_atoms(atoms);
+     * @endcode
+     */
+    void get_all_atoms(HandleSeq& appendToHandles) const
+    {
+        // Defer to handles by type for ATOMs and subclasses.
+        get_handles_by_type(appendToHandles, ATOM, true);
```

Comment:
OK, realize also that a lot of code has suffered through multiple backwards-compat re-writes, and so as the API changed, the code that uses it did not. As a result, it tends to look as if its badly written.  Some of it was, from the very begining; the worst offenders have now been removed; they were in the embodiment module.  So when you see poor code, there's often a historical reason for that.